libr/util/str.c

```diff
@@ -3972,3 +3972,19 @@ R_API bool r_str_startswith(const char *str, const char *needle) {
 	return !strncmp (str, needle, strlen (needle));
 }
 
+
+R_API const char *r_str_skip(const char *s, int n) {
+	int i;
+	r_return_val_if_fail (n < 0, "");
+	for (i = 0; i < n && *s != 0; i++) {
+		s++;
+	}
+	return s;
```

Comment:
imho this function should return NULL if you want to seek beyond the end of the string. shouldn it?

Reply:
The user can already use a `strlen ()` comparison to check if an index if out of bounds.

For text parsing, it usually simplifies code to stay in non-null territory. An empty string and an invalid string sometimes take the same code paths in an algorithm. That's why I thought it'd be a good idea to add this helper.

Semantically, the developer would want to express "give me the chars N bytes back". Then, an empty string indicates "there are no chars". Since text parsers work on char streams, they don't care whether that's due to OOB or because the string really ended. This is why Python supports sub-slicing out-of-bounds and Go equates `len([]byte{})` and `len(nil)`.

We could annotate the return parameter with `R_NONNULL` to make the API more obvious.